documentation/API-Testing-Guidelines.md

```diff
@@ -1,41 +1,288 @@
 # API Testing Guidelines
 
 ## Table of Contents
-1. [Introduction](#introduction)
-2. [Test Cases contributions for APIs](#contribution)
-3. [Location of feature file](#location)
-4. [Best practices and recommendations](#recommendations)
-7. [References](#references)
+- [Introduction](#introduction)
+- [Test plan coverage](#coverage)
+  - [Independent operations](#independent-operations)
+  - [CRUD operations](#crud-operations)
+  - [Notifications](#notifications)
+  - [Error scenarios](#error-scenarios)
+- [Test plan design guidelines](#guidelines)
+  - [Location of feature file](#location)
+  - [Feature structure](#feature-structure)
+  - [Environment variables](#environment-variables)
+  - [Background section](#background)
+  - [Scenario structure](#scenario-structure)
+  - [Request setup](#request-setup)
+  - [Request sending](#request-sending)
+  - [Response validation](#response-validation)
+- [References](#references)
 
 ## Introduction <a name="introduction"></a>
 This document captures guidelines for the API testing in CAMARA project. These guidelines are applicable to every API to be worked out under the CAMARA initiative.
 
-## Test Cases contribution for APIs <a name="contribution"></a>
-Based on the decision taken in the Commonalities working group and as documented in [API-Readiness-Checklist](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Readiness-Checklist.md), at least one Gherkin feature file will be added to the main subproject repo and this will fulfill the minimum criteria of readiness with respect to API test cases and documentation. If subprojects also intend to add test implementations, an aligned single implementation that is agreed among all provider implementors could be added to the main subproject repo. If no alignment is possible, each provider implementer will add the test implementation to their own repos.
+## Test plan coverage <a name="coverage"></a>
 
-## Location of feature file <a name="location"></a>
-The feature file will reside under 
-```
-Subproject_Repository/code/Test_definitions/file_name.feature
-for e.g. https://github.com/camaraproject/QualityOnDemand/blob/main/code/Test_definitions/QoD_API_Test.feature
-```
+Based on the decision taken in the Release Management and Commonalities working groups, and as documented in [API-Readiness-Checklist](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Readiness-Checklist.md), at least one Gherkin feature file will be added to the main subproject repo and this will fulfill the minimum criteria of readiness with respect to API test cases and documentation:
```

Comment:
I propose to change the link to Release Management wiki.
Current API Readiness Checklist will be deprecated: https://github.com/camaraproject/Commonalities/issues/202
```suggestion
Based on the decision taken in the Release Management and Commonalities working groups, and as documented in [API Release Process](https://wiki.camaraproject.org/display/CAM/API+Release+Process)](https://wiki.camaraproject.org/display/CAM/API+Release+Process#APIReleaseProcess-APIreadinesschecklist), at least one Gherkin feature file will be added to the main subproject repo and this will fulfill the minimum criteria of readiness with respect to API test cases and documentation:
```